src/middlewares/optional_query_param.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Middleware that extracts a typed optional query parameter from the HTTP request.
+ */
+import * as t from "io-ts";
+
+import { right } from "fp-ts/lib/Either";
+import { IRequestMiddleware } from "italia-ts-commons/lib/request_middleware";
+import {
+  IResponseErrorValidation,
+  ResponseErrorFromValidationErrors
+} from "italia-ts-commons/lib/responses";
+
+export function OptionalQueryParamMiddleware<S, A>(
+  name: string,
+  type: t.Type<A, S>
+): IRequestMiddleware<"IResponseErrorValidation", A | undefined> {
+  return async request => {
+    if (request.query[name] === undefined) {
+      return right(undefined);
+    }
+    return type
+      .decode(request.query[name])
+      .mapLeft<IResponseErrorValidation>(
+        ResponseErrorFromValidationErrors(type)
+      );
+  };
+}
```

Comment:
Great job, well done. I see you copied from https://github.com/pagopa/io-functions-commons/blob/214722423d701e8a8a508cf635ce4ebf7d47fc53/src/utils/middlewares/optional_query_param.ts#L19, it's ok as that package references an incompatible version of `fp-ts`.